src/backend/cuda/kernel/join.hpp

```diff
@@ -73,6 +73,15 @@ namespace cuda
                         blocksPerMatY * X.dims[3],
                         1);
 
+            int maxBlocksY = cuda::getDeviceProp(cuda::getActiveDeviceId()).maxGridSize[1];
+            // Increase the work per block until the condition is satisfied
+            // Increase threads only upto 512
+            while(blocks.y > maxBlocksY) {
```

Comment:
I had a similar version first https://github.com/arrayfire/arrayfire/pull/1641/commits/fda93803120241e8fd923b00cf5826e91e7f199a. It did not take into account that if blocks.y remains greater than maxBlocksY, the error will remain. Hence the while loop.